src/main/webapp/app/course/competencies/competency-management/competency-management-table.component.ts

```diff
@@ -0,0 +1,135 @@
+            this.pluralCompetencyType = 'prerequisites';
+        }
+    }
+
+    ngOnDestroy() {
+        this.dialogErrorSource.unsubscribe();
+    }
+
+    /**
+     * Opens a modal for selecting a course to import all competencies from.
+     */
+    openImportAllModal() {
+        const modalRef = this.modalService.open(ImportAllCompetenciesComponent, { size: 'lg', backdrop: 'static' });
+        //unary operator is necessary as otherwise courseId is seen as a string and will not match.
+        modalRef.componentInstance.disabledIds = [+this.courseId];
+        modalRef.componentInstance.competencyType = this.competencyType;
+        modalRef.result.then((result: ImportAllFromCourseResult) => {
+            const courseTitle = result.courseForImportDTO.title ?? '';
+
+            this.service
+                .importAll(this.courseId, result.courseForImportDTO.id!, result.importRelations)
+                .pipe(
+                    filter((res: HttpResponse<Array<CompetencyWithTailRelationDTO>>) => res.ok),
+                    map((res: HttpResponse<Array<CompetencyWithTailRelationDTO>>) => res.body),
+                )
+                .subscribe({
+                    next: (res: Array<CompetencyWithTailRelationDTO>) => {
+                        if (res.length > 0) {
+                            this.alertService.success(`artemisApp.${this.competencyType}.importAll.success`, { noOfCompetencies: res.length, courseTitle: courseTitle });
+                            this.updateDataAfterImportAll(res);
+                        } else {
+                            this.alertService.warning(`artemisApp.${this.competencyType}.importAll.warning`, { courseTitle: courseTitle });
+                        }
+                    },
+                    error: (res: HttpErrorResponse) => onError(this.alertService, res),
+                });
+        });
+    }
+
+    /**
+     * Updates the component and its relation chart with the new data from the importAll modal
+     * @param res Array of DTOs containing the new competencies and relations
+     * @private
+     */
+    updateDataAfterImportAll(res: Array<CompetencyWithTailRelationDTO>) {
+        const importedCompetencies = res.map((dto) => dto.competency).filter((element): element is CourseCompetency => !!element);
+
+        const importedRelations = res
+            .map((dto) => dto.tailRelations)
+            .flat()
+            .filter((element): element is CompetencyRelationDTO => !!element)
+            .map((dto) => dtoToCompetencyRelation(dto));
+        this.courseCompetencies = this.courseCompetencies.concat(importedCompetencies);
+        this.relations = this.relations.concat(importedRelations);
+    }
+
+    /**
+     * Delete a competency (and its relations)
+     *
+     * @param competencyId the id of the competency
+     */
+    deleteCompetency(competencyId: number) {
+        this.service.delete(competencyId, this.courseId).subscribe({
```

Comment:
same as above, we could add this to an subscription and unsubscribe

Reply:
As far as I can see, we usually don't do it for other rest calls as well. Is there a good reason, why we should start now?

Reply:
yes your right we rather focus on other reasons to unsubscribe 👍